### PR TITLE
build: upgrade Go installation

### PR DESCRIPTION
**Pull request by antonioiubatti93**
- Bump Go 1.15 in `go.mod`
- use Go 1.17 in setup action
- test against 1.15, 1.16 and 1.17

### CODE REVIEW

**Comment by antonioiubatti93** [comment]
[Upgrade services to Go 1.17](https://trello.com/c/qJvzrrfv/1471-upgrade-services-to-go-117)

**Review by caroline105 — APPROVED** [review]

File: .github/workflows/golangci.yml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.16.x
+          go-version: 1.17.x
 
       - name: Update dependencies
-        run: go mod tidy
+        run: go mod tidy -go=1.17
```
Comment (vthiery) [inline .github/workflows/golangci.yml:17]: We could keep it backward compatible and keep testing against `[1.15.x, 1.16.x, 1.17.x]` (2 "old" versions are probably enough) + bump the go version in go.mod. No?

**Review by vthiery — CHANGES_REQUESTED** [review]

File: .github/workflows/golangci.yml (hunk 1 above)
Comment (antonioiubatti93) [reply .github/workflows/golangci.yml:17]: That's exactly the advice I was looking for, thx :blush: I did the same for private repos, but I wasn't sure whether 1.17 only was ok for this public one. I'll do.

> bump the go version in go.mod

You mean, using the earliest go version among the test ones (in this case, Go 1.15)?

Comment (vthiery) [reply .github/workflows/golangci.yml:17]: Yes, the version in go.mod (and the ones more recent) should be tested IMO

**Comment by antonioiubatti93** [comment]
@vthiery shall I proceed with this one?

**Review by vthiery — APPROVED** [review]

**Pull request merged**